lib/types/urlencoded.js

```diff
@@ -19,18 +19,76 @@ var debug = require('debug')('body-parser:urlencoded')
 var deprecate = require('depd')('body-parser')
 var read = require('../read')
 var typeis = require('type-is')
+var qs = require('qs')
 
 /**
  * Module exports.
  */
 
 module.exports = urlencoded
 
+var charsetBySentinel = {
+  // This is what browsers will submit when the ✓ character occurs in an
+  // application/x-www-form-urlencoded body and the encoding of the page containing
+  // the form is iso-8859-1, or when the submitted form has an accept-charset
+  // attribute of iso-8859-1. Presumably also with other charsets that do not contain
+  // the ✓ character, such as us-ascii.
+  '%26%2310003%3B': 'iso-8859-1', // encodeURIComponent('&#10003;')
+  // These are the percent-encoded utf-8 octets representing a checkmark, indicating
+  // that the request actually is utf-8 encoded.
+  '%E2%9C%93': 'utf-8' // encodeURIComponent('✓')
+}
+
```

Comment:
this does confuse me a bit tho; it kind of seems like code that already exists in qs? meaning that the options can just be passed to qs and it'll handle the logic.

altho if someone's using a different querystring parser then the logic might need replicating

Reply:
You're correct that these magic values are also found [here](https://github.com/ljharb/qs/blob/f90cc35dd65c7099c35ae75d7a1a67aab85220e1/lib/parse.js#L49-L52). I spent some time trying to remember why also having the decoding logic here was warranted, but it seems like [we can get actually get rid of it](https://github.com/papandreou/body-parser/commit/cbf4cec5c601473c25226d205fcaa84c57706308) and still have the test suite pass!

I wonder if I did it like this because it seemed like a long shot to get changes into the `qs` module, so I went for `body-parser` first, because that's what I needed myself, if I recall correctly. It has been 6 years, though, so I'm not really sure 🤔